src/Parallel/ParallelComponentHelpers.hpp

```diff
@@ -80,6 +80,27 @@ using get_const_global_cache_tags_from_pdal =
             get_action_list_from_phase_dep_action_list<tmpl::_1>>>,
         Parallel_detail::get_const_global_cache_tags_from_action<tmpl::_1>>>>;
 
+namespace detail {
+template <typename PhaseAction>
+struct get_initialization_actions_list {
+  using type = tmpl::list<>;
+};
+
+template <typename PhaseType, typename InitializationActionsList>
+struct get_initialization_actions_list<Parallel::PhaseActions<
+    PhaseType, PhaseType::Initialization, InitializationActionsList>> {
+  using type = InitializationActionsList;
+};
+}  // namespace detail
+
+/// \ingroup ParallelGroup
+/// \brief Given the phase dependent action list, return the list of
+/// actions in the Initialization phase (or an empty list if the Initialization
+/// phase is absent from the phase dependent action list)
+template <typename PhaseDepActionList>
+using get_initialization_actions_list = tmpl::flatten<tmpl::transform<
```

Comment:
Why not use `tmpl::find(_if)` here instead? If there's a good reason I think that's worth documenting since that's how I would expect the algorithm to work.

Reply:
Please explain what you are thinking as there is little documentation of `tmpl::find` and I don't see any for `_if`.

Reply:
after discussion with @nilsdeppe, will leave current implementation as using `tmpl::find` would require that the list of PhaseActions always include the initialization phase even when it has an empty action list.